sdcm/scan_operation_thread.py

```diff
@@ -174,16 +327,9 @@ class FullPartitionScanThread(ScanOperationThread):
     reversed_query_filter_ck_by = {'lt': ' and {} < {}', 'gt': ' and {} > {}', 'lt_and_gt': ' and {} < {} and {} > {}',
                                    'no_filter': ''}
 
-    def __init__(self, **kwargs):
-        super().__init__(scan_event=FullPartitionScanReversedOrderEvent, **kwargs)
-        self.full_partition_scan_params = kwargs
-        self.full_partition_scan_params['validate_data'] = json.loads(
-            self.full_partition_scan_params.get('validate_data', 'false'))
-        self.pk_name = self.full_partition_scan_params.get('pk_name', 'pk')
-        self.ck_name = self.full_partition_scan_params.get('ck_name', 'ck')
-        self.data_column_name = self.full_partition_scan_params.get('data_column_name', 'v')
-        self.rows_count = self.full_partition_scan_params.get('rows_count', 5000)
-        self.table_clustering_order = self.get_table_clustering_order()
+    def __init__(self, generator, **kwargs):
+        super().__init__(generator, scan_event=FullPartitionScanReversedOrderEvent, **kwargs)
```

Comment:
seems like this is only a leftover, but why in `FullPartitionScanOperation` the default `scan_event` is `FullPartitionScanReversedOrderEvent` ?

Reply:
Because that's the first event type that will be used in this class (and only one if we don't do data validaiton). First query is the "reversed" one, than a validation query is made with the "normal" query at witch point the `scan_event` param is changed to a regular `FullScanEvent`.